.clang-tidy

```diff
@@ -36,6 +36,9 @@
 #  -performance-avoid-endl: we would like to turn this on, but there are too
 #      many legitimate uses in our samples.
 #
+#  -readability-identifier-naming: We want to use invalid case style for a class
```

Comment:
Other than `google::cloud::future` and friends  what other classes have this problem? Feels like we should be able to fix these with a small number of `NOLINT`.

I guess we can merge and fix afterwards too.